packages/core/src/components/Tag/Tag.tsx

```diff
@@ -94,34 +94,38 @@ export const HvTag = (props: HvTagProps) => {
     ...others
   } = useDefaultProps("HvTag", props);
   const { activeTheme, selectedMode } = useTheme();
+  const { classes, cx, css } = useClasses(classesProp);
 
   const getDeleteIcon = () => {
     const disabledSemanticColor =
       type === "semantic" ? "secondary_60" : "base_dark";
     const { tabIndex = 0 } = deleteButtonProps;
 
+    const closeIconStyles = css({
+      ...(disabled ? { cursor: "not-allowed" } : undefined),
+      height: 16,
+      "& svg .color0": {
+        fill: theme.colors[disabled ? disabledSemanticColor : "base_dark"],
+      },
+    });
     return (
-      <StyledButton
+      <HvButton
         classes={{
-          startIcon: clsx(tagClasses.tagButton, classes?.tagButton),
-          focusVisible: clsx(tagClasses.focusVisible, classes?.focusVisible),
-          root: clsx(tagClasses.button, classes?.button),
+          startIcon: classes.tagButton,
+          focusVisible: classes.focusVisible,
+          root: classes.primaryButton,
```

Comment:
We should continue to use the `button` class instead of `primaryButton` since it's a breaking change.